tsdb/head_append.go

```diff
@@ -605,6 +610,11 @@ func (s *memSeries) appendHistogram(t int64, sh histogram.SparseHistogram, appen
 		return sampleInOrder, chunkCreated
 	}
 
+	if value.IsStaleNaN(sh.Sum) {
+		// Deliberately empty other fields so that any future histogram will result in a new chunk.
+		sh = histogram.SparseHistogram{Sum: sh.Sum}
+	}
```

Comment:
Do we really need this? It seems the only way of receiving a stale marker here is via line 270 above, and that's already an empty histogram (with the exception of Sum).

Reply:
I don't want to rule out the possibility of getting it via headAppender.AppendHistogram() directly and cause regression later. That said, I think it is enough to take care of this in the histoChunk.AppendHistogram().